sdk/storage/azure-storage-blob/tests/test_container.py

```diff
@@ -2467,6 +2467,24 @@ def test_download_blob(self, **kwargs):
 
         assert downloaded.readall() == data
 
+    @BlobPreparer()
+    @recorded_by_proxy
+    def test_download_blob_unicode_boundary(self, **kwargs):
+        storage_account_name = kwargs.pop("storage_account_name")
+        storage_account_key = kwargs.pop("storage_account_key")
+
+        bsc = BlobServiceClient(self.account_url(storage_account_name, "blob"), storage_account_key)
+        bsc._config.max_single_get_size = 1
```

Comment:
This is on to force chunking, should be left out to test non-chunking path

Reply:
Note: `read` does **not** work properly at the moment if it is a ranged encode read with chunking... Issue being: the stream doesn't read enough and there is nothing in the buffer (`current_content`) 😄